BlazarService/src/main/java/com/hubspot/blazar/util/BuildConfigUtils.java

```diff
@@ -0,0 +1,117 @@
+package com.hubspot.blazar.util;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.base.Optional;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.hubspot.blazar.base.BuildConfig;
+import com.hubspot.blazar.base.BuildStep;
+import com.hubspot.blazar.base.Dependency;
+import com.hubspot.blazar.base.GitInfo;
+import com.hubspot.blazar.base.PostBuildSteps;
+import com.hubspot.blazar.base.StepActivationCriteria;
+import com.hubspot.blazar.config.BlazarConfiguration;
+import com.hubspot.blazar.config.ExecutorConfiguration;
+import com.hubspot.blazar.exception.NonRetryableBuildException;
+import com.hubspot.blazar.external.models.singularity.Resources;
+
+@Singleton
+public class BuildConfigUtils {
+
+  public static final String BUILDPACK_FILE = ".blazar-buildpack.yaml";
+  private static final Logger LOG = LoggerFactory.getLogger(BuildConfigUtils.class);
+  private final ExecutorConfiguration executorConfiguration;
+  private final GitHubHelper gitHubHelper;
+
+  @Inject
+  public BuildConfigUtils(BlazarConfiguration blazarConfiguration,
+                          GitHubHelper gitHubHelper) {
+    this.executorConfiguration = blazarConfiguration.getExecutorConfiguration();
+    this.gitHubHelper = gitHubHelper;
+  }
+
+  public BuildConfig getConfigForBuildpackOnBranch(GitInfo gitInfo) throws IOException, NonRetryableBuildException {
+    return getConfigAtRefOrDefault(gitInfo, BUILDPACK_FILE, gitInfo.getBranch());
+  }
+
+  public BuildConfig ensureDefaultConfigurationForBuild(BuildConfig buildConfig) {
```

Comment:
I think `addMissingBuildConfigSettings` could be more easy to understand than `ensureDefaultConfigurationForBuild`